portal-impl/src/com/liferay/portlet/dynamicdatamapping/service/impl/DDMTemplateLocalServiceImpl.java

```diff
@@ -1409,14 +1412,13 @@ protected String formatScript(String type, String language, String script)
 
 			try {
 				script = DDMXMLUtil.validateXML(script);
-				script = DDMXMLUtil.formatXML(script);
 			}
-			catch (Exception e) {
-				throw new TemplateScriptException();
+			catch (PortalException pe) {
+				throw new TemplateScriptException(pe);
 			}
 		}
 
-		return script;
+		return DDMXMLUtil.formatXML(script);
```

Comment:
you're right about the if check, it should be inside of it... but the systemException is a runtime exception in trunk so we don't need to catch it.